Fix check for secure flag in cookie

strehle opened the pull request:
found by accident

aramprice committed:
Use constants when accessing session contents

reedr3 committed:
ForcePasswordChangeController persists SecurityContext to session

This resolves an issue where SessionResetFilter does not see the updates
UaaAuthentication.authenticatedTime value

Signed-off-by: aram price <pricear@vmware.com>

strehle committed:
Fix check for secure flag in cookie

found by accident

strehle committed:
Fix check for secure flag in cookie

found by accident

aramprice committed:
Members can be final

aramprice committed:
Extract LDIF file URL loading to a const

aramprice committed:
Harmonize resource access patterns in test helpers

aramprice committed:
Merge pull request #1385 from cloudfoundry/ap/persist-security-context-in-force-password-flow

ForcePasswordChangeController persists SecurityContext to session

cf-identity committed:
Update UAA image reference in k8s deployment template to cfidentity/uaa@sha256:4258ca996c973ba21f0816a94e6fa1a4187374eddd489612d19bafc1ff43e8ff

aramprice committed:
Rename `scripts/uaa.log.sh` => `scripts/tail-uaa-log`

simplify implementation

cf-identity committed:
Update UAA image reference in k8s deployment template to cfidentity/uaa@sha256:37bc92882b247d2612aba6a8d41b8f4c7395800ad81c1bfe0c1a558a49c9c4cd

aramprice committed:
DRY up server starup in InMemoryLdapServer

Commit:
SPRING_SESSION_ATTRIBUTES is `MEDIUMBLOB` in MySQL

The Spring JDBC session attribute column in MySQL is `BLOB` which is
smaller (64Kb) than needed for certain UAA installations.

The column type in MySQL is changed to `MEDIUMBLOB` is 16Mb which will
hopefully provide enough space.

[fixes #172851963]

Signed-off-by: aram price <pricear@vmware.com>

aramprice committed:
Extract version to dependencies.gradle

aramprice committed:
Fix deprecation warnings for Gradle 7.0

shamus commented:
Good catch! It seems like some of the unit tests are failing.  Could you address those?

reedr3 committed:
Remove SessionPageIT

- it tests the deprecated `/session` endpoint on SessionController
- it is broken because of a change in Chrome's JS cross-origin behavior
- testing the non-deprecated `/session_management` to be added later

[finishes #174629302]

Signed-off-by: aram price <pricear@vmware.com>

cf-identity committed:
Update UAA image reference in k8s deployment template to cfidentity/uaa@sha256:bf4d07063baf8d0ba63048cd9443d0cfef247d87cea83d5283fe8911aef48542

aramprice committed:
Test methods are descriptive and java-conventional

aramprice committed:
Assertions are ordered to give better errors

cf-identity committed:
Update UAA image reference in k8s deployment template to cfidentity/uaa@sha256:ecdcb9a429a9827084dd0228fda44b7ac58b92153de853c1a2058a3264dfb296

aramprice committed:
Merge pull request #1389 from cloudfoundry/ap/dry-up-in-memory-ldap-server

DRY up server startup in InMemoryLdapServer

aramprice committed:
Merge pull request #1390 from cloudfoundry/ap/gradle-cleanup

Extract version into dependencies.gradle and fix some deprecation warnings

aramprice committed:
Merge pull request #1383 from cloudfoundry/ap/constantize-session-keys

Use constants when accessing session contents

cf-identity committed:
Update UAA image reference in k8s deployment template to cfidentity/uaa@sha256:1de2b01d4a94eea23513bf8c9e08810b89ea8dcbcb9dc2de68cd995345fab1cb

aramprice committed:
Use golang 1.15

Includes chages from `go mod tidy`

aramprice committed:
Merge pull request #1393 from cloudfoundry/ap/update-golang

Use golang 1.15 in `k8s/`

aramprice committed:
Update all k8s.io dependencies in lock-step

`k8s.io/client-go` has been locked to `v11.0.0+incompatible` for reasons
which are lost to time. this prevented `k8s.io/api` and `k8s.io/apimachinery` from being updated past `v0.18.8`.

cf-identity committed:
Update UAA image reference in k8s deployment template to cfidentity/uaa@sha256:1c6c7c5537960621f02d8ed0f07845a2e2d8aceabc38ca0e83758717c625fdf6

aramprice committed:
Merge pull request #1394 from cloudfoundry/ap/k8s.io-updates

Update all k8s.io dependencies in lock-step

cf-identity committed:
Update UAA image reference in k8s deployment template to cfidentity/uaa@sha256:a64493b253ef94deec7e9d3e60810b8e54cd8aed9281a3d3ba7ddc94af518435

strehle committed:
Merge branch 'fix/csrf_cookie' of github.com:cloudfoundry/uaa into fix/csrf_cookie

* 'fix/csrf_cookie' of github.com:cloudfoundry/uaa:
  Fix check for secure flag in cookie

strehle commented:
started again in #1398
